runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -203,6 +232,39 @@ private boolean checkCoreArchitecture() {
         return true;
     }
 
+    private boolean verifyPackageInfo(PackageInfo packageInfo,
+            String algorithm, String fingerprint) {
+        if (packageInfo.signatures == null) {
+            Log.d(TAG, "No signature in package info");
```

Comment:
As you said "we should be as vocal as we can", we already prepared a dialog in `XWalkActivity` for this kind of situation. Generally, I treat the failures in checking the signature as a whole, it means that just check the result of `verifyPackageInfo()` but not deal with the specific error.

The verification failure just means the runtime library is not officially released, not necessarily it's malicious. We have done our job to inform the user that the integrity cannot be verified and also prompt to uninstall it and get the correct one. What to do next is up to the user. If we raise a assertion error or something else during the verification, the subsequent user interaction would be breaked.